examples/advanced_operations/add_ad_customizer.py

```diff
@@ -0,0 +1,459 @@
+        customer_id: a client customer ID.
+        ad_group_ids: a list of ad group IDs.
+        feed_name: the name of the feed to create.
+    """
+    ad_group_service = client.get_service("AdGroupService", version="v5")
+    ad_group_ad_service = client.get_service("AdGroupAdService", version="v5")
+    ad_group_ad_operations = []
+
+    for ad_group_id in ad_group_ids:
+        ad_group_ad_operation = client.get_type(
+            "AdGroupAdOperation", version="v5"
+        )
+        ad_group_ad = ad_group_ad_operation.create
+        ad_group_ad.ad_group = ad_group_service.ad_group_path(
+            customer_id, ad_group_id
+        )
+        ad_group_ad.ad.final_urls.append("http://www.example.com")
+        ad_group_ad.ad.expanded_text_ad.headline_part1 = (
+            f"Luxury cruise to {{={feed_name}.Name}}"
+        )
+        ad_group_ad.ad.expanded_text_ad.headline_part2 = (
+            f"Only {{={feed_name}.Price}}"
+        )
+        ad_group_ad.ad.expanded_text_ad.description = (
+            f"Offer ends in {{=countdown({feed_name}.Date)}}!"
+        )
+        ad_group_ad_operations.append(ad_group_ad_operation)
+
+    try:
+        response = ad_group_ad_service.mutate_ad_group_ads(
+            customer_id, ad_group_ad_operations
+        )
+        print(f"Added {len(response.results)} ads:")
+        for ad in response.results:
+            print(f"Added an ad with resource name {ad.resource_name}")
+    except GoogleAdsException as ex:
+        _handle_google_ads_exception(ex)
+
+
+def _handle_google_ads_exception(exception):
+    """Prints the details of a GoogleAdsException object.
+
+    Args:
+        exception: an instance of GoogleAdsException.
+    """
+    print(
+        f'Request with ID "{exception.request_id}" failed with status '
+        f'"{exception.error.code().name}" and includes the following errors:'
+    )
+    for error in exception.failure.errors:
+        print(f'\tError with message "{error.message}".')
+        if error.location:
+            for field_path_element in error.location.field_path_elements:
+                print(f"\t\tOn field: {field_path_element.field_name}")
+    sys.exit(1)
+
+
+if __name__ == "__main__":
+    # GoogleAdsClient will read the google-ads.yaml configuration file in the
+    # home directory if none is specified.
+    google_ads_client = GoogleAdsClient.load_from_storage()
+
+    parser = argparse.ArgumentParser(
+        description=(
+            "Adds an ad customizer feed and associates it with a customer."
+        )
+    )
+    # The following argument(s) should be provided to run the example.
+    parser.add_argument(
+        "-c",
+        "--customer_id",
+        type=str,
+        required=True,
+        help="The Google Ads customer ID.",
+    )
+    parser.add_argument(
+        "-a",
+        "--ad_group_ids",
+        nargs=2,
+        type=str,
+        required=True,
+        default=[],
```

Comment:
see comment line 340. Should not be both `required=True` and `default=[]`

Reply:
Good call, I'll remove the default param.